language/Type-Classes.md

```diff
@@ -183,8 +183,101 @@ Currently, instances for the following classes can be derived by the compiler:
 - Data.Generic.Rep (class Generic) 
 - [Data.Eq (class Eq)](https://pursuit.purescript.org/packages/purescript-prelude/docs/Data.Eq#t:Eq)
 - [Data.Ord (class Ord)](https://pursuit.purescript.org/packages/purescript-prelude/docs/Data.Ord#t:Ord)
+- [Data.Foldable (class Foldable)](https://pursuit.purescript.org/packages/purescript-foldable-traversable/docs/Data.Foldable#t:Foldable)
 - [Data.Functor (class Functor)](https://pursuit.purescript.org/packages/purescript-prelude/docs/Data.Functor#t:Functor)
 - [Data.Newtype (class Newtype)](https://pursuit.purescript.org/packages/purescript-newtype/docs/Data.Newtype#t:Newtype)
+- [Data.Traversable (class Traversable)](https://pursuit.purescript.org/packages/purescript-foldable-traversable/docs/Data.Traversable#t:Traversable)
+
+#### `Functor`, `Foldable`, and `Traversable`
+
+All three of these classes can be derived for certain data types, provided the following requirements are met:
+
+1. As these classes all have kind `(Type -> Type) -> Constraint`, the data type must have at least one type parameter.
+2. Every occurrence of the final parameter in the data type must be in a legal position. Legal positions are:
+   * Data constructor arguments
+   * A field in a record type that itself appears in a legal position
+   * The rightmost argument to a type constructor that itself appears in a legal position. In this case, the type constructor must have an instance of the same class being derived. (For example, function types returning the parameter can be used when deriving `Functor` but not when deriving `Foldable` or `Traversable`.)
+
+(Note for Haskell programmers: unlike the very similar feature in GHC, PureScript does not currently consider the `a` in `((a -> b) -> b)` to be a legal position in the above sense when deriving `Functor`.)
```

Comment:
Maybe this point should also be documented in the "difference from Haskell" page?

Reply:
Good question, no strong opinions from me.

Reply:
Done (the list of GHC extensions supported by PureScript needed updating anyway).